test/arrays.js

```diff
@@ -137,6 +137,33 @@
     equal(_.sortedIndex(array, 2147483648), 2147483648, 'should work with large indexes');
   });
 
+  test('sortedIndex conforms to sortBy for undefined (#1834)', function() {
+    var sorted = _.sortBy([undefined, 1, undefined, 2]);
+    equal(_.sortedIndex(sorted, undefined, _.identity), 2);
+
+    var edgeCaseNumbers = [-Infinity, -Infinity, 0, Number.MAX_VALUE, Infinity, Infinity, undefined, undefined, NaN];
```

Comment:
Number.MAX_VALUE won't exist in all environments. Other than that we'll have to discuss the implications. May be worth while to add #1882 at the same time